filecoin-proofs/tests/paramfetch/prompts_to_fetch.rs

```diff
@@ -146,3 +155,50 @@ fn fetches_vk_even_if_sector_size_does_not_match() -> Result<(), FailureError> {
 
     Ok(())
 }
+
+#[test]
+fn invalid_json_path_produces_error() -> Result<(), FailureError> {
+    let mut session = ParamFetchSessionBuilder::new(Some(PathBuf::from("/invalid/path")))
+        .with_session_timeout_ms(1000)
+        .build();
+
+    session.exp_string("fatal error: json file '/invalid/path' does not exist")?;
+
+    Ok(())
+}
+
+#[test]
+fn invalid_json_produces_error() -> Result<(), FailureError> {
+    let manifest_pbuf = tmp_manifest(None)?;
+
+    let mut file = File::create(&manifest_pbuf)?;
+    file.write_all(b"invalid json")?;
+
+    let mut session = ParamFetchSessionBuilder::new(Some(manifest_pbuf))
+        .with_session_timeout_ms(1000)
+        .build();
+
+    session.exp_string("fatal error: json file")?;
+    session.exp_string("did not parse correctly: expected value at line 1 column 1")?;
+
+    Ok(())
+}
+
+#[test]
+fn no_json_path_uses_default_manifest() -> Result<(), FailureError> {
+    let file = File::open("../parameters.json")?;
+    let reader = BufReader::new(file);
+    let manifest: ParameterMap = serde_json::from_reader(reader)?;
+
+    let mut session = ParamFetchSessionBuilder::new(None)
+        .with_session_timeout_ms(1000)
+        .build();
+
+    session.exp_string("using built-in manifest")?;
+
+    for parameter in manifest.keys() {
```

Comment:
If the order of the keys in `manifest` is nondeterministic, will this test occasionally fail? I don't recall how `session.exp_string` works.

Reply:
afaict the order of the keys is deterministic. `BTreeMap` is a sorted map